beetsplug/fetchart.py

```diff
@@ -317,13 +325,36 @@ def get(self, album, plugin, paths):
         """Return the Cover Art Archive and Cover Art Archive release group URLs
         using album MusicBrainz release ID and release group ID.
         """
+        candidate_url = ""
         if 'release' in self.match_by and album.mb_albumid:
-            yield self._candidate(url=self.URL.format(mbid=album.mb_albumid),
-                                  match=Candidate.MATCH_EXACT)
+            candidate_url = self.URL.format(mbid=album.mb_albumid)
+            match_type = Candidate.MATCH_EXACT
         if 'releasegroup' in self.match_by and album.mb_releasegroupid:
-            yield self._candidate(
-                url=self.GROUP_URL.format(mbid=album.mb_releasegroupid),
-                match=Candidate.MATCH_FALLBACK)
+            candidate_url = self.GROUP_URL.format(
+                            mbid=album.mb_releasegroupid)
+            match_type = Candidate.MATCH_FALLBACK
```

Comment:
This seems to produce a candidate for only _one_ of the release or release group covers, whereas previously it would yield _both_ candidates (and so we would try both).

Maybe a way to preserve the old behavior would be to yield all four candidates: the two full-sized images, and the two thumbnail-sized URLs. I _think_ if the the thumbnails candidates are yielded first, then they will be tried first and used if they exist. This could also avoid the need for an explicit `HEAD` request, because the centralized logic that tries all URLs in a cascade and uses the first "working" one would kick in.

Reply:
Interesting, thanks. I'll try that out.

Reply:
I didn't see this comment until after I'd already pushed the previous change, so I'll put this PR back in draft status until I work this out.

Reply:
@sampsyo Please review PR #3647 instead. I think it's much simpler.